lib/genome/online_updaters/docm/updater.rb

```diff
@@ -0,0 +1,132 @@
+require 'net/http'
+module Genome; module OnlineUpdaters; module Docm
+  class Updater
+    attr_reader :new_version
+    def initialize(source_db_version = Date.today.strftime("%d-%B-%Y"))
+      @new_version = source_db_version
+    end
+
+    def update
+      remove_existing_source
+      source = create_new_source
+      create_interaction_claims(source)
+    end
+
+    private
+    def create_interaction_claims(source)
+      api_client = ApiClient.new
+      api_client.variants.each do |variant|
+        interaction_information = parse_interaction_information(variant)
+        interaction_information.each do |interaction_info|
+          gc = get_or_create_gene_claim(variant, source)
+          dc = get_or_create_drug_claim(interaction_info, source)
+          get_or_create_interaction_claim(gc, dc, interaction_info, variant['diseases'], source)
+        end
+      end
+    end
+
+    def get_or_create_gene_claim(variant, source)
+      DataModel::GeneClaim.where(
+        source_id: source.id,
+        nomenclature: 'DoCM Entrez Gene Symbol',
+        name: variant['gene']
+      ).first_or_create
+      #Entrez Gene Id
+    end
+
+    def parse_interaction_information(variant)
+      if !variant.key?('meta')
+        return []
+      end
+      fields = variant['meta'].first['Drug Interaction Data']['fields']
+      rows = variant['meta'].first['Drug Interaction Data']['rows']
+      interaction_information = Set.new()
+      rows.each do |row|
+        row_hash = {}
+        fields.zip(row).each do |name, value|
+          row_hash[name] = value
+        end
+        row_hash['Therapeutic Context'].split(/,|\+|plus/).each do |drug|
+          info = row_hash
+          info['Therapeutic Context'] = drug
```

Comment:
This works, but not how you're expecting. `info` will just point to `row_hash`, so you're just overwriting the original value each time i.e.

```
irb(main):001:0> x = { 1 => 2, 3 => 4}
=> {1=>2, 3=>4}
irb(main):002:0> y = x
=> {1=>2, 3=>4}
irb(main):003:0> y[1] = 5
=> 5
irb(main):004:0> x
=> {1=>5, 3=>4}
```

Adding it to the set does copy it rather than add a pointer:

```
irb(main):003:0> z = Set.new
=> #<Set: {}>
irb(main):004:0> x =  {1 => 2, 3 => 4}
=> {1=>2, 3=>4}
irb(main):005:0> y = x
=> {1=>2, 3=>4}
irb(main):006:0> z.add(x)
=> #<Set: {{1=>2, 3=>4}}>
irb(main):007:0> y[1] = 5
=> 5
irb(main):008:0> y
=> {1=>5, 3=>4}
irb(main):009:0> x
=> {1=>5, 3=>4}
irb(main):010:0> z.add(x)
=> #<Set: {{1=>5, 3=>4}, {1=>5, 3=>4}}>
```
So the code ends up working, but if you're just going to overwrite the original value each time, there's no need for the local `info` variable. Alternately you could could use `dup` which copies the hash which would behave the way I think you intended:

```
irb(main):001:0> x = { 1 => 2, 3 => 4}
=> {1=>2, 3=>4}
irb(main):002:0> y = x.dup
=> {1=>2, 3=>4}
irb(main):003:0> y[1] = 5
=> 5
irb(main):004:0> y
=> {1=>5, 3=>4}
irb(main):005:0> x
=> {1=>2, 3=>4}
```